src/app/services/user-service/user.service.ts

```diff
@@ -102,10 +110,115 @@ export class UserService {
 
 	}
 
-	// add user method
-	addUser(user :User) :Observable<User> {
-		return this.http.post<User>(this.url, user, {headers: this.headers});
-	}
+	// adds user if form is filled out and user input matches a real address
+	async addUser(user :User) : Promise<User> {
+        const addressValid = await this.addressValidation(user);//waits for validation to edit fields before post request
+        console.log(`User location info log(location modifying to '' on failure) happens first: ${user.hAddress}, ${user.hCity}, ${user.hState}, ${user.hZip}`)
+        return this.http.post<User>(this.url, user, {headers: this.headers}).toPromise();
+    }
+
+    async addressValidation(user:User): Promise<any> {
```

Comment:
We need to make this function a bit more maintainable. 

- [ ] This is a high level function that has a lot of constituent steps.  The steps should be broken into their own (private) functions that are then called by this function.  

- [ ] Each function should be thoroughly documented using doc comments: /** */ rather than /* */. Comments within the function look fairly thorough, just give the high level doc comments on each one so people know the intent of each function.

- [ ] Once they are broken down and we can show that they work, we should remove the comments as these will be use visible.

- [ ] Fix linting errors. Install ts-lint if you don't have it installed in VS Code already.